### PR TITLE
osd/dmclock/testing: reorganize testing, building now optional

### PR DESCRIPTION
**Pull request by ivancich**
Modify cmake files to take advantage or re-organization of dmclock's
cmake set-up, specifically not having dmclock's cmake files call
add_test. Remove dmclock tests from being dependencies on ceph's
"test" target. Make change so dmclock tests are not built unless
WITH_TESTS and WITH_DMCLOCK_TESTS are both set. This is so openSUSE
Leap will build correctly.

### CODE REVIEW

**Comment by ivancich** [comment]
Trying to get the dmclock testing fixes in sooner rather than later to help the FreeBSD and OpenSUSE folk, so splitting these commits from dmclock integration part 2.

File: src/dmclock/cmake/modules/FindGTest.cmake [hunk 1]
```diff
@@ -2,27 +2,30 @@
 #
 #  GTEST_INCLUDE_DIRS   - where to find mcas/mcas.h, etc.
 #  GTEST_LIBRARIES      - List of libraries when using mcas.
+#  GTEST_MAIN_LIBRARIES - List of main libraries when using mcas.
```
Comment (tchaikov) [inline src/dmclock/cmake/modules/FindGTest.cmake]: the same applies to `FindGTest.cmake`.

File: src/dmclock/support/src/run_every.cc [hunk 2]
```diff
@@ -1,11 +1,11 @@
 // -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
 // vim: ts=8 sw=2 smarttab
 /*
- * Copyright (C) 2016 Red Hat Inc.
+ * Copyright (C) 2017 Red Hat Inc.
  */
 
 
-#include <iostream>
+// #include <iostream>
```
Comment (tchaikov) [inline src/dmclock/support/src/run_every.cc]: nit, maybe we can just drop this line, if we don't want it at all.

File: src/dmclock/cmake/modules/FindGTest.cmake (hunk 1 above)
Comment (ivancich) [inline src/dmclock/cmake/modules/FindGTest.cmake]: I wasn't aware the cmake versions 2.8.11 and later included these. I will remove them and update the PR.

File: src/dmclock/support/src/run_every.cc (hunk 2 above)
Comment (ivancich) [inline src/dmclock/support/src/run_every.cc]: I will clean this up and update the PR. Thanks, Kefu.

**Comment by liewegas** [comment]
retest this please

**Comment by smithfarm** [comment]
Related PR: #15174

**Pull request merged**